sys/include/net/coap.h

```diff
@@ -163,14 +163,20 @@ extern "C" {
  * @name    CoAP method codes used in header
  * @{
  */
-#define COAP_CLASS_REQ          (0)
-#define COAP_METHOD_GET         (1)
-#define COAP_METHOD_POST        (2)
-#define COAP_METHOD_PUT         (3)
-#define COAP_METHOD_DELETE      (4)
-#define COAP_METHOD_FETCH       (5)
-#define COAP_METHOD_PATCH       (6)
-#define COAP_METHOD_IPATCH      (7)
+#define COAP_CLASS_REQ          (0) /**< Code Class for Request */
+
+/**
+ * @brief   CoAP method codes used in request
+ */
+typedef enum {
+    COAP_METHOD_GET = 1,            /**< GET request (no paylod) */
+    COAP_METHOD_POST,               /**< POST request (resource processes payload) */
+    COAP_METHOD_PUT,                /**< PUT request (update resource with payload) */
+    COAP_METHOD_DELETE,             /**< DELETE request (no payload, remove resource)*/
+    COAP_METHOD_FETCH,              /**< FETCH request (RFC 8132) */
+    COAP_METHOD_PATCH,              /**< PATCH request (RFC 8132) */
+    COAP_METHOD_IPATCH,             /**< iPATCH request (RFC 8132) */
```

Comment:
I would prefer to see these all get a number explicitly assigned in the enum, to hint to the reader that the exact value of the enum matters